---
title: "Class 14:"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
counts <- read.csv("airway_scaledcounts.csv", stringsAsFactors = FALSE, row.names=1)
```

```{r}
metadata <-  read.csv("airway_metadata.csv", stringsAsFactors = FALSE)

```


```{r}
head(counts)
```


```{r}
head(metadata)
```


There are 'r nrow(counts)' genes in the 
```{r}
nrow(counts)
```

```{r}
colnames(counts)
```


```{r}
metadata$id
```

Lets check the correspondence of our countData cols and our metaData rows. This will be required 

```{r}
all(colnames(counts) == metadata$id)
```


## Section 3. Walk through of gene expression analysis 

Our metadata file tells us which columns of the countData are drug treated and control 
(i.e. non drugged)

## Control group 

```{r}
control.inds<- metadata$dex == "control"
control<- metadata[control.inds,]
control
```

```{r}
head(counts[, control$id])
```


```{r}
control.mean<- rowSums(counts[, control$id]) / length(control$id) 
```


## Treated group

```{r}
treated.inds<- metadata$dex == "treated"
treated<- metadata[treated.inds, ]
treated
```

```{r}
treated.mean<- rowSums(counts[, treated$id])/ length(treated$id)
```


Lets store control.mean and treated.mean together for ease of use
```{r}
meancounts<- data.frame(control.mean, treated.mean)
```


What is the sum counts across all the genes in control and treated
```{r}
colSums(meancounts)
```


```{r}
plot(meancounts)
```

Plot a log-log scale so we can see our data 
```{r}
plot(meancounts, log="xy")
```


```{r}
meancounts$log2fc <- log2(meancounts[,"treated.mean"]/meancounts[,"control.mean"])
head(meancounts)
```

```{r}
to.rm<- unique(which(meancounts[, 1:2] ==0, arr.ind = TRUE )[,1] )
```


```{r}
mycounts<- meancounts[-to.rm, ]
nrow(mycounts)
```

How many genes are up in the drug treated
```{r}
up.inds<- mycounts$log2fc >2
```

```{r}
sum(up.inds)
```


and how many are down?
```{r}
down.ind <- mycounts$log2fc < (-2)
```

```{r}
sum(down.inds)
```

```{r}
head(mycounts[up.inds,])
```


## Section 4. Adding annotation data

We can add annotation from a supplied CSV file

```{r}
anno<- read.csv("annotables_grch38.csv")
head(anno)
```


Use the **(merge)** function to add the annotation data from the `anno` object 
to our RNA-Seq results in `my counts`
```{r}
# use the merge function with mycounts and anno 
mycounts.anno<- merge(mycounts, anno, by.x="row.names", by.y="ensgene")
```

```{r}
mycounts.anno
```

```{r}
library("AnnotationDbi")
library("org.Hs.eg.db")
```

```{r}
columns(org.Hs.eg.db)
```


We will use the **mapIDs(*)** from bioconductor to add annotation data to our results 

```{r}
mycounts$symbol<-  mapIds(org.Hs.eg.db, 
                   keys=row.names(mycounts), #The names in our dataset
                  keytype = "ENSEMBL",  #Format of our gene names
                  column = "SYMBOL" )   #What new data we want to add
```

```{r}
head(mycounts)
```



```{r}
mycounts$entrez<- mapIds(org.Hs.eg.db, 
                   keys=row.names(mycounts), #The names in our dataset
                  keytype = "ENSEMBL",  #Format of our gene names
                  column = "ENTREZID" )   #What new data we want to add
```

```{r}
head(mycounts)
```


## Section 5. Use DESeq2

```{r}
library("DESeq2")
```



```{r}
counts <- read.csv("airway_scaledcounts.csv", stringsAsFactors = FALSE)
metadata <-  read.csv("airway_metadata.csv", stringsAsFactors = FALSE)
```

```{r}
dds <- DESeqDataSetFromMatrix(countData=counts, 
                              colData=metadata, 
                              design=~dex, 
                              tidy=TRUE)
dds
```

```{r}
dds<- DESeq(dds)
```

```{r}
res <- results(dds)
res
```

```{r}
summary(res)
```

```{r}
res05<- results(dds, alpha=0.05)
summary(res05)
```

Store our results as a data.frame object
```{r}
res01<- as.data.frame(results(dds, alpha = 0.01))
head(res01)
```


Let's sort or order our results by the adjusted p-value 

```{r}
# an example of using order to sort a vector
x<- c(5,4,1,2)
x[order (x)]
```

```{r}
ord.inds<- order(res01$padj)
write.csv(res01[ord.inds, ], file = "signif01_results.csv")
```


```{r}
mycols<- rep("gray", nrow(res01))
mycols[res01$padj <0.01] <- "red"
mycols[ abs(res01$log2FoldChange) >2] <- "blue"

mycols[ (res01$padj < 0.01) & (abs(res01$log2FoldChange) >2)] <- "red"

plot(res01$log2FoldChange, -log(res01$padj), 
     xlab= "Fold Change (Drug/ Control)", 
     ylab= "-log(P-value)", 
     col=mycols)
```











